File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;
        int[] output = new int[n];
        // let's count the number of zeros
        int zeros_count = 0;
        int prod = 1;
        for (int x : nums){
            if (x==0){
                zeros_count++;
            }else{
                prod = prod * x;
            }
        }
        if (zeros_count>1){
            return output;
        }

        for (int i = 0;i<n;i++){
            if (zeros_count==1){
                if (nums[i]==0){
                    output[i] = prod;
                }else{
                    output[i] = 0;
                }
            }else{
                output[i]  = prod/nums[i];
            }
        }
        return output;


    }
}  

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        n = len(nums)
        hm = {}
        size = 0
        for e in nums:
            depth = 0
            v = e -1
            while v in hm:
                v-=1
                depth+=1
            v = e+1
            while v in hm:
                v+=1
                depth+=1
            size = max (size ,depth+1)
            if e not in hm:
                hm[e] = 1
        return size
                